src/array.c

```diff
@@ -476,7 +477,8 @@ JL_DLLEXPORT jl_value_t *jl_pchar_to_string(const char *str, size_t len)
 
 JL_DLLEXPORT jl_value_t *jl_alloc_string(size_t len)
 {
-    jl_value_t *s = jl_gc_alloc(jl_get_ptls_states(), sizeof(size_t)+len+1, jl_string_type);
+    size_t sz = sizeof(size_t) + len + 1; // add space for trailing \nul protector and size
+    jl_value_t *s = jl_gc_alloc_(jl_get_ptls_states(), sz, jl_string_type); // force inlining
```

Comment:
These are starting to get fairly repetitive. Maybe factor out into a common subroutine?

Reply:
there's only two of them, so it's not worthwhile

Reply:
Also, this method technically is that common subroutine. The code is duplicated in `jl_pchar_to_string` for performance.